plugins/Ntuplizer.cc

```diff
@@ -522,211 +553,247 @@ Ntuplizer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
-    //	  LeptonWeight    *= theElectronAnalyzer->GetElectronIdSF(TightElecVect.at(1), 0);
-    //
-    //	  //LeptonWeightUnc += pow(theElectronAnalyzer->GetElectronRecoEffSFError(ElecVect.at(0))   ,2);
-    //	  //LeptonWeightUnc += pow(theElectronAnalyzer->GetElectronRecoEffSFError(ElecVect.at(1))   ,2);
-    //	  //LeptonWeightUnc += pow(theElectronAnalyzer->GetElectronIdSFError(ElecVect.at(0), 0)     ,2);
-    //	  //LeptonWeightUnc += pow(theElectronAnalyzer->GetElectronIdSFError(ElecVect.at(1), 0)     ,2);
-    //
-    //	  //LeptonWeightUp   = LeptonWeight+sqrt(LeptonWeightUnc);
-    //	  //LeptonWeightDown = LeptonWeight-sqrt(LeptonWeightUnc);
-    //	}
+    	  LeptonWeightUp   = LeptonWeight+sqrt(LeptonWeightUnc);
+	  LeptonWeightDown = LeptonWeight-sqrt(LeptonWeightUnc);
+    	}
 
 
-    // }
-    //}
+    }
+    }
 
-    //else if(isWtoMN) {
+    else if(isZtoEE) {
+      if(isVerbose) std::cout << "Do the Z->e e" << std::endl;
+     int e1(-1), e2(-1);
+         for(unsigned int i = 0; i < TightElecVect.size(); i++) {
+    	for(unsigned int j = 1; j < TightElecVect.size(); j++) {
+    	  if(i==j || TightElecVect[i].charge() == TightElecVect[j].charge())
+    	    {
+    	      isZtoEE = false;
+    	      continue;
+    	    }
+    	  float Zmass = (TightElecVect[i].p4() + TightElecVect[j].p4()).mass();
+    	  if(Zmass > 50. && Zmass < 130.)
+    	    {
+    	      e1 = i;
+    	      e2 = j;
+    	      isZtoEE =true;
+    	    }
+    	}
+    }
+	 //    Build candidate
+    if(e1 >= 0 && e2 >= 0) {
+    	theZ.addDaughter(TightElecVect.at(e1).charge() ? TightElecVect.at(e1) : TightElecVect.at(e2));
+    	theZ.addDaughter(TightElecVect.at(e1).charge() ? TightElecVect.at(e2) : TightElecVect.at(e1));
+    	addP4.set(theZ);
+    	isZtoEE = true;
+
+
+    	// SF
+    	if(isControl && isMC && !is2016) {
+    	  float LeptonWeightUnc = 0.;
+	  LeptonWeightUp = 0.;
+	  LeptonWeightDown = 0.;
+    	  /// FIXME -> APPLYING THE SF FOR Ele27Tight HADRCODED <- FIXME ///
+    	  // if (TightElecVect.at(e1).pt() > TightElecVect.at(e2).pt() ){
+    	  //   LeptonWeight     *= theElectronAnalyzer->GetElectronTriggerSFEle27Tight(TightElecVect.at(e1));
+    	  //   //LeptonWeightUnc  += pow(theElectronAnalyzer->GetElectronTriggerSFErrorEle27Tight(ElecVect.at(e1)),2);
+    	  // }
+    	  // else{
+    	  //   LeptonWeight     *= theElectronAnalyzer->GetElectronTriggerSFEle27Tight(TightElecVect.at(e2));
+    	  //   //LeptonWeightUnc  += pow(theElectronAnalyzer->GetElectronTriggerSFErrorEle27Tight(ElecVect.at(e2)),2);
+    	  // }
+    	  LeptonWeight    *= theElectronAnalyzer->GetElectronRecoEffSF(TightElecVect.at(0));
+    	  LeptonWeight    *= theElectronAnalyzer->GetElectronRecoEffSF(TightElecVect.at(1));
+    	  LeptonWeight    *= theElectronAnalyzer->GetElectronIdSF(TightElecVect.at(0), 0);
+    	  LeptonWeight    *= theElectronAnalyzer->GetElectronIdSF(TightElecVect.at(1), 0);
```

Comment:
Same comment as for muons: Vector filled with tight electrons (`id=3`) but SF computed for `id=0`

Reply:
Same here, tight id = 3.

Reply:
Just to be sure, we should also chage the indices to `e1` and `e2` here. Suggestion:
```suggestion
    	  LeptonWeight    *= theElectronAnalyzer->GetElectronIdSF(TightElecVect.at(e1), 3);
     	  LeptonWeight    *= theElectronAnalyzer->GetElectronIdSF(TightElecVect.at(e2), 3);
```